src/prefs-dialog.c

```diff
@@ -1797,7 +1804,10 @@ prefs_dialog_init (PrefsDialog *dialog)
   setup_fonts_page (dialog);
   setup_stored_data_page (dialog);
   setup_language_page (dialog);
-  if (mode != EPHY_EMBED_SHELL_MODE_APPLICATION)
+
+  if (mode == EPHY_EMBED_SHELL_MODE_APPLICATION || mode == EPHY_EMBED_SHELL_MODE_INCOGNITO)
```

Comment:
I think this is not enough. Because sync relies on global GSettings, it should probably only be enabled in browser mode. E.g. we do not want synced data leaking into private profile mode, right?